src/runtime/runtime_stm32_timers.go

```diff
@@ -67,26 +79,27 @@ func ticks() timeUnit {
 
 // Enable the timer used to count ticks
 func initTickTimer(ti *timerInfo) {
+
 	tickTimer = ti
 	ti.EnableRegister.SetBits(ti.EnableFlag)
 
-	psc := uint32(TICK_TIMER_FREQ / TICK_RATE)
-	period := uint32(1)
+	// Compute pre-scaler
+	timerPSC = uint32(TICK_TIMER_CLOCKFREQ / TICK_TIMER_INTFREQ)
+	timerARR = uint32(1)
 
-	// Get the pre-scale into range, with interrupt firing
-	// once per tick.
-	for psc > 0x10000 || period == 1 {
-		psc >>= 1
-		period <<= 1
+	// Find correct prescaler/Counter limit
+	for timerARR < 512 {
```

Comment:
Why 512?  If it's to get precision in ARR, might want to do something more like this? https://github.com/tinygo-org/tinygo/pull/1867/files#diff-45e7f001091fdbec2fb598052e4dd8232744d1a2974bd2e658a29c874e8e5855R124

(borrowed from another PWM implementation - it chooses PSC to get max precision in ARR)